docs/topics/native/native-ios-integration.md

```diff
@@ -0,0 +1,252 @@
+[//]: # (title: iOS integration)
+
+> This page describes the new memory manager enabled by default since Kotlin 1.7.20. See our [Migration guide](native-migration-guide.md)
+> to move your projects from the legacy memory manager.
+>
+{type="note"}
+
+Integration of Kotlin/Native garbage collector with Swift/Objective-C ARC is seamless and generally requires no additional
+work to be done. Learn more about [Swift/Objective-C interoperability](native-objc-interop.md).
+
+However, there are some specifics you should keep in mind:
+
+## Threads
+
+### Deinitilizers
+
+Deinit on the Swift/Objective-C objects and the objects they refer to is called on a different thread if
+these objects cross interop boundaries into Kotlin/Native, for example:
+
+```Swift
```

Comment:
These samples are mixes of Kotlin and Swift code. It is better to separate them to improve readability.

Reply:
Thank you for noticing! broke into separate code blocks